core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -2424,8 +2424,12 @@ private void scheduleAsyncPersistenceInternal(LockedInodePath inodePath) throws
   public FileSystemCommand workerHeartbeat(long workerId, List<Long> persistedFiles)
       throws FileDoesNotExistException, InvalidPathException, AccessControlException {
     for (long fileId : persistedFiles) {
-      // Permission checking for each file is performed inside setAttribute
-      setAttribute(getPath(fileId), SetAttributeOptions.defaults().setPersisted(true));
+      try {
+        // Permission checking for each file is performed inside setAttribute
+        setAttribute(getPath(fileId), SetAttributeOptions.defaults().setPersisted(true));
+      } catch (FileDoesNotExistException | AccessControlException | InvalidPathException e) {
+        LOG.error("Failed to set file {} as persisted, because {}", fileId, e);
+      }
```

Comment:
I see, let's just fix this issue here. We can consider some strategies for protecting an async persist file from metadata changes in the future or resolving the conflicts with the ufs.